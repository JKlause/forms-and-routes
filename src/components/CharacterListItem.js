import React from 'react';
import PropTypes from 'prop-types';

const CharacterListItem = ({ photoUrl, name, _id, handleCharacterClick }) => (
  <div onClick={() => handleCharacterClick(_id)}>
    <img src={photoUrl} />
    <figcaption>{name}</figcaption>
  </div>

);

CharacterListItem.propTypes = {
  photoUrl: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  _id: PropTypes.object.isRequired,
  handleCharacterClick: PropTypes.func.isRequired
};


export default CharacterListItem;
