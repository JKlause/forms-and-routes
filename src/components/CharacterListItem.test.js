import React from 'react';
import { shallow } from 'enzyme';
import CharacterListItem from './CharacterListItem';

describe('CharacterListItem component', () => {
  const props = {
    photoUrl: '',
    name: '',
    _id: '',
    handleCharacterClick: () => {}
  };
  
  it('renders CharacterListItem', () => {   
    const wrapper = shallow(<CharacterListItem {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
