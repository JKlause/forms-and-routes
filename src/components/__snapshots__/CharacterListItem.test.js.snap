// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterListItem component renders CharacterListItem 1`] = `
<div
  onClick={[Function]}
>
  <img
    src=""
  />
  <h2 />
</div>
`;
